src/components/PopupWithForm.js

```diff
@@ -25,11 +25,20 @@ export default class PopupWithForm extends Popup {
     }
   }
 
+  //   // Find the submit button in the form by class name
+  // const submitButton = updateAvatarForm.querySelector(".modal__button");
+  // // Update the text before making the API call
+  // submitButton.textContent = "Saving...";
+
+  setButtonText(buttonText) {
```

Comment:
Great that you've implemented the logic for rendering loading via the class methods!